### PR TITLE
[qa] feefilter: Clear mempool after each check

### PR DESCRIPTION
**Pull request by maflcko**
Reported by @sdaftuar in https://github.com/bitcoin/bitcoin/pull/7148#issuecomment-226680429

### CODE REVIEW

File: qa/rpc-tests/p2p-feefilter.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def allInvsMatch(invsExpected, testnode):
             if (sorted(invsExpected) == sorted(testnode.txinvs)):
                 return True;
         time.sleep(1)
-    return False;
+    raise AssertionError("Did not match; Timed out");
```
Comment (laanwj) [inline qa/rpc-tests/p2p-feefilter.py]: nit: No need for a `;`

Comment (MarcoFalke) [inline qa/rpc-tests/p2p-feefilter.py]: Changed to `.`.

**Comment by MarcoFalke** [comment]
This fixes the following travis failure:

```py
  File "/home/travis/build/bitcoin/bitcoin/build/../qa/rpc-tests/p2p-feefilter.py", line 101, in run_test
    assert(allInvsMatch(txids, test_node))

**Comment by sdaftuar** [comment]
Thanks for trying to fix this.  This might be good enough, but it occurred to me that in the situation where clearing the mempool is preventing the test failure we're trying to fix, it's really just covering up the fact that the test isn't exercising the code path we want to test.  

The current test failures occur when the time from mempool-acceptance-to-relay exceeds 10 seconds (which it sometimes does, as the delay is random).  So if we clear the mempool after 10 seconds to avoid this scenario, then we're not really exercising the feefilter logic in that situation.

So I think a more robust fix would be forget about clearing the mempool, and instead send a single transaction that should be received regardless of filter, and syncing with the receipt of that inv.  I think we only need to do this in the test case where we call `allInvsMatch` and expect nothing (at line 96).  We could then also get rid of the `time.sleep(10)` at line 95.

ping @morcos

**Pull request closed without merging**